Sources/joyfill-components-swift/Components/Table/CollectionView/CollectionViewTable.swift

```diff
@@ -94,13 +74,35 @@ public class CollectionViewTable: UIView, UIImagePickerControllerDelegate & UINa
         updateRowNumber()
         registerCollectionViewCell()
     }
+    
+    func registerCollectionViewCell() {
+        addSubview(collectionView)
+        collectionView.register(CollectionViewCell.self, forCellWithReuseIdentifier: "Cell")
+        
+        //Add constraint
+        collectionView.leadingAnchor.constraint(equalTo: leadingAnchor).isActive = true
+        collectionView.trailingAnchor.constraint(equalTo: trailingAnchor).isActive = true
+        collectionView.topAnchor.constraint(equalTo: topAnchor).isActive = true
+        collectionView.bottomAnchor.constraint(equalTo: bottomAnchor).isActive = true
+        
+        numberOfColumns = tableColumnType.count
+        tableHeading = tableColumnTitle
+        
+        // To show fake rows with empty data
+        if valueData.count == 1 {
+            valueData.append(emptyValueElement)
+            valueData.append(emptyValueElement)
+        } else if valueData.count == 2 {
+            valueData.append(emptyValueElement)
+        } else {}
+    }
 }
 
 // Table class extension to add collectionView delegate and dataSource methods
 extension CollectionViewTable: UICollectionViewDelegate, UICollectionViewDataSource, UICollectionViewDelegateFlowLayout {
     // MARK: CollectionView delegate method for number of sections
     public func numberOfSections(in collectionView: UICollectionView) -> Int {
-        return numberOfRows
+        return 4
```

Comment:
Is this set to 4 due to it being in the Document view are before they click on it to edit it? Otherwise we shouldn't restrict the number of rows

Reply:
In the field (1st screen of table) numberOfRows are fixed as we discussed. After click on view when modal will appear there all the rows will come.